sdk/include/reactos/shlwapi_undoc.h

```diff
@@ -324,6 +324,22 @@ IContextMenu_Invoke(
 
 DWORD WINAPI SHGetObjectCompatFlags(IUnknown *pUnk, const CLSID *clsid);
 
+/* Flags for appending an extension in PathFileExistsDefExtAndAttributesW */
+#define PADE_PIF      0x0001
+#define PADE_COM      0x0002
+#define PADE_EXE      0x0004
+#define PADE_BAT      0x0008
+#define PADE_LNK      0x0010
+#define PADE_CMD      0x0020
+#define PADE_OPTIONAL 0x0040
+#define PADE_ALL (PADE_PIF | PADE_COM | PADE_EXE | PADE_BAT | PADE_LNK | PADE_CMD | PADE_OPTIONAL)
```

Comment:
These flags just look like the already-existing ones https://git.reactos.org/?p=reactos.git;a=blob;f=sdk/include/reactos/shlwapi_undoc.h;hb=044f18195031bf45e3ec7e17ec32041cae8f331d#l286
and you can notice that they should be related (or just the same), because:
`/* dwWhich flags for PathFileExistsDefExtW and PathFindOnPathExW */`
--> notice the `PathFileExistsDefExt`

So the `PathFileExistsDefExtAndAttributesW` function is an extension to the `PathFileExistsDefExtW` one.
(More precisely, I think that the `PathFileExistsDefExtW` should be implementing by calling the more general `PathFileExistsDefExtAndAttributesW` (in order to avoid duplicated code), and just drop the attributes away.)

Reply:
Fixed and reduced code.

Reply:
Tested successfully.